src/Nethermind/Nethermind.Consensus/ExecutionRequests/ExecutionRequestProcessor.cs

```diff
@@ -0,0 +1,142 @@
+// SPDX-FileCopyrightText: 2024 Demerzel Solutions Limited
+// SPDX-License-Identifier: LGPL-3.0-only
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Nethermind.Abi;
+using Nethermind.Core;
+using Nethermind.Core.ExecutionRequest;
+using Nethermind.Core.Extensions;
+using Nethermind.Core.Specs;
+using Nethermind.Crypto;
+using Nethermind.Evm;
+using Nethermind.Evm.Tracing;
+using Nethermind.Evm.TransactionProcessing;
+using Nethermind.Int256;
+using Nethermind.State;
+
+namespace Nethermind.Consensus.ExecutionRequests;
+
+public class ExecutionRequestsProcessor(ITransactionProcessor transactionProcessor) : IExecutionRequestsProcessor
+{
+    private readonly AbiSignature _depositEventABI = new("DepositEvent", AbiType.DynamicBytes, AbiType.DynamicBytes, AbiType.DynamicBytes, AbiType.DynamicBytes, AbiType.DynamicBytes);
+    private readonly AbiEncoder _abiEncoder = AbiEncoder.Instance;
+    private const int depositRequestsBytesSize = 48 + 32 + 8 + 96 + 8;
+    private const int withdrawalRequestsBytesSize = 20 + 48 + 8;
+    private const int consolidationRequestsBytesSize = 20 + 48 + 48;
+
+    private const long GasLimit = 30_000_000L;
+
+    public IEnumerable<ExecutionRequest> ProcessDeposits(TxReceipt[] receipts, IReleaseSpec spec)
+    {
+        if (!spec.DepositsEnabled)
+            yield break;
+
+        for (int i = 0; i < receipts.Length; i++)
+        {
+            LogEntry[]? logEntries = receipts[i].Logs;
+            if (logEntries is not null)
+            {
+                for (var j = 0; j < logEntries.Length; j++)
+                {
+                    LogEntry log = logEntries[j];
+                    if (log.LoggersAddress == spec.DepositContractAddress)
+                    {
+                        yield return DecodeDepositRequest(log);
+                    }
+                }
+            }
+        }
+    }
+
+    private ExecutionRequest DecodeDepositRequest(LogEntry log)
+    {
+        object[] result = _abiEncoder.Decode(AbiEncodingStyle.None, _depositEventABI, log.Data);
+        List<byte> flattenedResult = new List<byte>();
```

Comment:
The previous classes were better optimized. I think we should be able to do it better, avoid list allocations etc. Think about it